google_guest_agent/windows_accounts.go

```diff
@@ -108,18 +108,12 @@ func printCreds(creds *credsJSON) error {
 	return err
 }
 
-var badExpire []string
-
 func (k windowsKey) expired() bool {
-	t, err := time.Parse(time.RFC3339, k.ExpireOn)
+	expired, err := utils.CheckExpired(k.ExpireOn)
 	if err != nil {
-		if !containsString(k.ExpireOn, badExpire) {
-			logger.Errorf("error parsing time: %s", err)
-			badExpire = append(badExpire, k.ExpireOn)
-		}
-		return true
+		logger.Debugf(err.Error())
```

Comment:
OK, so I've gone back and forth on this.
First, this won't actually be logged at that rate, since we are only logging the error on Debugf and not Errorf.

So, that means that there is a change here. We wouldn't be logging bad expire times to the user. However, that's the behavior of all of the other key errors currently. We don't log any of the errors to user output except for an error parsing the time. I'm not sure how useful logging just that one case is.

Reply:
OK, This has been updated. In our offline conversation we discussed what "consistent" was in regards to logging to Debug if a bad key has already been seen. The old code would print a message to debug every time (although without details), so we thought consistent meant to log to Error if we hadn't seen that bad key and log to Debug if we had. However, in looking through the code some more, I noticed that we do a similar thing with bad Windows keys, and in that case nothing gets printed to Debug, so I'd argue that consistent means just logging to Error the first time and don't try to log any additional errors with bad keys to Debug.